src/execution_plan/ops/op_conditional_traverse.c

```diff
@@ -18,62 +18,12 @@ static OpResult CondTraverseReset(OpBase *opBase);
 static OpBase *CondTraverseClone(const ExecutionPlan *plan, const OpBase *opBase);
 static void CondTraverseFree(OpBase *opBase);
 
-static void _setupTraversedRelations(CondTraverse *op, QGEdge *e) {
-	uint reltype_count = array_len(e->reltypeIDs);
-	if(reltype_count > 0) {
-		array_clone(op->edgeRelationTypes, e->reltypeIDs);
-		op->edgeRelationCount = reltype_count;
-	} else {
-		op->edgeRelationCount = 1;
-		op->edgeRelationTypes = array_new(int, 1);
-		op->edgeRelationTypes = array_append(op->edgeRelationTypes, GRAPH_NO_RELATION);
-	}
-}
-
-// Updates query graph edge.
-static int _CondTraverse_SetEdge(CondTraverse *op, Record r) {
-	// Consumed edges connecting current source and destination nodes.
-	if(!array_len(op->edges)) return 0;
-
-	Edge *e = op->edges + (array_len(op->edges) - 1);
-	Record_AddEdge(r, op->edgeRecIdx, *e);
-	array_pop(op->edges);
-	return 1;
-}
-
-// Collect edges between the source and destination nodes.
-static void __CondTraverse_CollectEdges(CondTraverse *op, int src, int dest) {
-	Node *srcNode = Record_GetNode(op->r, src);
-	Node *destNode = Record_GetNode(op->r, dest);
-	for(int i = 0; i < op->edgeRelationCount; i++) {
-		Graph_GetEdgesConnectingNodes(op->graph,
-									  ENTITY_GET_ID(srcNode),
-									  ENTITY_GET_ID(destNode),
-									  op->edgeRelationTypes[i],
-									  &op->edges);
-	}
-}
-
-// Collect edges between the source and destination nodes matching the op's traversal direction.
-static void _CondTraverse_CollectEdges(CondTraverse *op, int src, int dest) {
-	switch(op->direction) {
-	case GRAPH_EDGE_DIR_OUTGOING:
-		__CondTraverse_CollectEdges(op, op->srcNodeIdx, op->destNodeIdx);
-		return;
-	case GRAPH_EDGE_DIR_INCOMING:
-		// If we're traversing incoming edges, swap the source and destination.
-		__CondTraverse_CollectEdges(op, op->destNodeIdx, op->srcNodeIdx);
-		return;
-	case GRAPH_EDGE_DIR_BOTH:
-		// If we're traversing in both directions, collect edges in both directions.
-		__CondTraverse_CollectEdges(op, op->srcNodeIdx, op->destNodeIdx);
-		__CondTraverse_CollectEdges(op, op->destNodeIdx, op->srcNodeIdx);
-		return;
-	}
+static inline int CondTraverseToString(const OpBase *ctx, char *buf, uint buf_len) {
```

Comment:
Although this function is basically a proxy function, it doesn't call too often for it to be declared as `inline`